edvart/plots.py

```diff
@@ -68,77 +68,135 @@ def scatter_plot_2d(
-        layout = dict(
-            width=figsize[0] * _INCHES_TO_PIXELS,
-            height=figsize[1] * _INCHES_TO_PIXELS,
-            xaxis=dict(showgrid=False, showticklabels=show_xticks, zeroline=show_zerolines),
-            yaxis=dict(showgrid=False, showticklabels=show_yticks, zeroline=show_zerolines),
-            legend=dict(title=f"<b>{color_col}</b>"),
-        )
 
+def _scatter_plot_2d_noninteractive(
+    df: pd.DataFrame,
+    x: Union[str, pd.Series, np.ndarray],
+    y: Union[str, pd.Series, np.ndarray],
+    color_col: Optional[str] = None,
+    figsize: Tuple[float, float] = (12, 12),
+    opacity: float = 0.8,
+    xlabel: Optional[str] = None,
+    ylabel: Optional[str] = None,
+    show_xticks: bool = False,
+    show_yticks: bool = False,
+    show_zerolines: bool = False,
+) -> None:
+    _fig, ax = plt.subplots(figsize=figsize)
     if color_col is not None:
         is_color_categorical = utils.is_categorical(df[color_col]) or not is_numeric(df[color_col])
+        if is_color_categorical:
+            color_categorical = pd.Categorical(df[color_col])
+            color_codes = color_categorical.codes
+        else:
+            color_codes = df[color_col]
+        scatter = ax.scatter(x, y, c=color_codes, alpha=opacity)
 
-        if interactive:
-            fig = go.Figure(layout=layout)
-            if is_color_categorical:
-                df = df.copy()
-                x_name, y_name = "__edvart_scatter_x", "__edvart_scatter_y"
-                df[x_name] = x
-                df[y_name] = y
-                for group_name, group in df.groupby(color_col):
-                    fig.add_trace(
-                        go.Scatter(
-                            x=group[x_name],
-                            y=group[y_name],
-                            mode="markers",
-                            marker=dict(opacity=opacity),
-                            name=group_name,
-                            text=[
-                                "</br>".join(
-                                    f"{col_name}: {df.loc[row, col_name]}"
-                                    for col_name in group.columns.drop([x_name, y_name])
-                                )
-                                for row in group.index
-                            ],
-                        )
-                    )
-            else:
+        if is_color_categorical:
+            legend_elements = scatter.legend_elements()
+            ax.legend(legend_elements[0], color_categorical.categories, title=color_col)
+        else:
+            cbar = plt.colorbar(scatter)
+            cbar.ax.set_ylabel(color_col)
+    else:
+        ax.scatter(x, y, alpha=opacity)
+
+    if xlabel is not None:
+        ax.set_xlabel(xlabel)
+    if ylabel is not None:
+        ax.set_ylabel(ylabel)
+    if not show_xticks:
+        ax.set_xticks([])
+    if not show_yticks:
+        ax.set_yticks([])
+    if not show_zerolines:
+        ax.grid(False)
```

Comment:
This is not present in the old version. Isn't it a new feature?

Reply:
You are right. I called it a fix since it the behavior should have been there before.
https://github.com/datamole-ai/edvart/pull/43